src/WaveTrack.cpp

```diff
@@ -1158,14 +1162,13 @@ void WaveTrack::HandleClear(double t0, double t1,
                   auto leftClip = std::make_unique<WaveClip>(*clip, mpFactory, true);
                   //i18n-hint Name assigned to the first clip when splitting a wave track
                   leftClip->SetName(XC("%s 1", "clip name template").Format(leftClip->GetName()).Translation());
-                  leftClip->Clear(t0, clip->GetEndTime());
+                  leftClip->TrimRight(clip->GetPlayEndTime() - t0);
                   clipsToAdd.push_back(std::move(leftClip));
 
                   auto rightClip = std::make_unique<WaveClip>(*clip, mpFactory, true);
                   //i18n-hint Name assigned to the second clip when splitting a wave track
                   rightClip->SetName(XC("%s 2", "clip name template").Format(rightClip->GetName()).Translation());
-                  rightClip->Clear(clip->GetStartTime(), t1);
-                  rightClip->Offset(t1 - clip->GetStartTime());
+                  rightClip->SetTrimLeft(t1 - rightClip->GetPlayStartTime());
```

Comment:
I think TrimLeft was intended, not SetTrimLeft:  else, clearing the middle of a clip, which already had some left trimming, will give wrong results for the right hand new clip.

Am I correct?

Reply:
Yes, thanks!